editors/code/src/util.ts

```diff
@@ -82,3 +82,30 @@ export function isRustDocument(document: vscode.TextDocument): document is RustD
 export function isRustEditor(editor: vscode.TextEditor): editor is RustEditor {
     return isRustDocument(editor.document);
 }
+
+export interface Disposable {
+    dispose(): void;
+}
+
+export function dispose(disposable: undefined | Disposable | Disposable[] | Map<unknown, Disposable>) {
+    if (disposable === undefined) return;
+
+    if (disposable instanceof Array) {
+        while (disposable.length > 0) {
+            dispose(disposable.pop());
```

Comment:
Disposing should always be wrapped in try/catch, so that panic in one dtor does not prevent other from firing. 

See a fun bug in std about this issue: https://github.com/rust-lang/rust/pull/67235

Reply:
This is a recursive call where try/catch is inside of it

Reply:
Ahh.... yeah, this is exactly my point about control flow being non-obvious :D

Reply:
Do you want inline it here and in Map case?

Reply:
I don't fully understand why we need anything beyond the original loop in the command.

Reply:
Just less boilerplate code... If you don't like I'll remove